src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -389,19 +389,10 @@ public RestoreCommand(IApplicationEnvironment env)
                     return;
                 }
 
-                // "dnu restore" is case-sensitive
                 if (!string.Equals(node.Item.Match.Library.Name, node.LibraryRange.Name, StringComparison.Ordinal))
                 {
-                    if (missingItems.Add(node.LibraryRange))
-                    {
-                        var errorMessage = string.Format("Unable to locate {0} {1}. Do you mean {2}?",
-                            node.LibraryRange.Name.Red().Bold(), node.LibraryRange.VersionRange, node.Item.Match.Library.Name.Bold());
-                        ErrorMessages.GetOrAdd(projectJsonPath, _ => new List<string>()).Add(errorMessage);
-                        Reports.Error.WriteLine(errorMessage);
-                        success = false;
-                    }
-
-                    return;
+                    // Fix casing of the library name to be installed
+                    node.Item.Match.Library.Name = node.LibraryRange.Name;
```

Comment:
@davidfowl 
`node.Item.Match.Library.Name` is the package ID used by installation and lockfile. We correct its value to ID used in **runtime dependency graph**. So everything is agreeing on the ID used by runtime dependency graph.